WalletWasabi.Fluent/ViewModels/Wallets/Send/TransactionPreviewViewModel.cs

```diff
@@ -22,75 +26,264 @@ public partial class TransactionPreviewViewModel : RoutableViewModel
+				SkipCommand = ReactiveCommand.CreateFromTask(async () => await OnConfirmAsync(_transaction));
+
 				NextCommand = ReactiveCommand.CreateFromTask(async () =>
 				{
-					var saved = await TransactionHelpers.ExportTransactionToBinaryAsync(transaction);
+					var saved = await TransactionHelpers.ExportTransactionToBinaryAsync(_transaction);
 
 					if (saved)
 					{
 						Navigate().To(new SuccessViewModel("The PSBT has been successfully created."));
 					}
 				});
+
 				_nextButtonText = "Save PSBT file";
 			}
 			else
 			{
-				NextCommand = ReactiveCommand.CreateFromTask(async () => await OnConfirmAsync(transaction));
+				NextCommand = ReactiveCommand.CreateFromTask(async () => await OnConfirmAsync(_transaction));
+
 				_nextButtonText = "Confirm";
 			}
 
+			AdjustFeeCommand = ReactiveCommand.CreateFromTask(async () =>
+			{
+				var feeRateDialogResult = await NavigateDialogAsync(new SendFeeViewModel(wallet, info, false));
+
+				if (feeRateDialogResult.Kind == DialogResultKind.Normal && feeRateDialogResult.Result != _info.FeeRate)
+				{
+					_info.FeeRate = feeRateDialogResult.Result;
+
+					var newTransaction = await BuildTransactionAsync();
+
+					if (newTransaction is { })
+					{
+						UpdateTransaction(newTransaction);
+					}
+				}
+			});
+
+			AvoidChangeCommand = ReactiveCommand.CreateFromTask(async () =>
+			{
+				var optimisePrivacyDialog =
+					await NavigateDialogAsync(new OptimisePrivacyViewModel(wallet, info, _transaction!));
+
+				if (optimisePrivacyDialog.Kind == DialogResultKind.Normal && optimisePrivacyDialog.Result is { })
+				{
+					UpdateTransaction(optimisePrivacyDialog.Result);
+				}
+			});
 		}
 
 		public bool PreferPsbtWorkflow => _wallet.KeyManager.PreferPsbtWorkflow;
 
-		public string AmountText { get; }
-
 		public string AddressText { get; }
 
-		public string FeeText { get; }
-
 		public string? PayJoinUrl { get; }
 
 		public bool IsPayJoin { get; }
 
+		public ICommand AdjustFeeCommand { get; }
+
+		public ICommand AvoidChangeCommand { get; }
+
+		private async Task<bool> InitialiseTransactionAsync()
+		{
+			var transactionInfo = _info;
+
+			var privacyControlDialogResult = await NavigateDialogAsync(new PrivacyControlViewModel(_wallet, transactionInfo, isSilent: true));
+			if (privacyControlDialogResult.Kind == DialogResultKind.Normal && privacyControlDialogResult.Result is { } coins)
+			{
+				transactionInfo.Coins = coins;
+			}
+			else if (privacyControlDialogResult.Kind != DialogResultKind.Normal)
+			{
+				return false;
+			}
+
+			var feeDialogResult = await NavigateDialogAsync(new SendFeeViewModel(_wallet, transactionInfo, true));
+			if (feeDialogResult.Kind == DialogResultKind.Normal)
+			{
+				transactionInfo.FeeRate = feeDialogResult.Result;
+			}
+			else
+			{
+				return false;
+			}
+
+			return true;
+		}
+
+		private async Task<BuildTransactionResult?> BuildTransactionAsync()
+		{
+			IsBusy = true;
+
+			var transactionInfo = _info;
```

Comment:
Same again, I think the reference is safe.

Reply:
done